### ISSUE TITLE
Monorepo support?

### ISSUE THREAD

**Issue by blackxored** [conventional-changelog/standard-version#225]
I've read a lot of threads without a conclusive answer. Is there monorepo support in standard-version or planned?

* Multiple changelog generator including a root changelog.
* Multiple package version bumps at once.
* etc

**Comment by stevemao** [comment]
lerna is already using `conventional-changelog`. I guess we should document it.

**Comment by bcoe** [comment]
@stevemao @blackxored; following up from @stevemao's comment. I helped add similar functionality to lerna, to the functionality in standard-version.

I think we'd be smart to reference lerna in our README for standard-version.

**Comment by blackxored** [comment]
Yes, I found lerna's `--conventional-commits` eventually. However, I think this package has a couple of more options. 
Lerna won't still give me a combined changelog at the root, unfortunately.

**Comment by pantoninho** [comment]
What's the state of this issue?

I don't understand if I should use lerna's option to create a `conventional-changelog` or use this package instead.

**Comment by iarroyo5** [comment]
@pantoninho I am using the one provided by `lerna`. At the moment with `conventional-changelog/standard-version` you can't have `independent` versions and generated `CHANGELOG.md` with the correct meta-info on them, because it bumps all the packages with the same version without respect the `independent` option.

For the first creation of the each _CHANGELOG.md` I use this [independent changelog generation tip](https://github.com/lerna/lerna/blob/d93510847a7ee37c56b196c0a5f32c746319b542/commands/version/README.md#tips)

And after it, having in your `lerna.json` file:

```
"command": {
    "version": {
       "message": "chore(release): publish",
       "conventionalCommits": true,
       "yes": true
    }
  }
```

you can simple call `npx lerna version` which will update your `CHANGELOG.md` files, will bump your package versions respecting them with the conventional commit preset chosen and will respect the `independent` versioning for each package.

**Comment by coder2000** [comment]
What if we are not using lerna? Is there still a way to use this with a monorepo?

**Comment by codeandcats** [comment]
Yes, I'm curious how I can use standard-version in a mono-repo (configured using yarn workspaces, not lerna) consisting of multiple npm packages.

**Comment by snebjorn** [comment]
Did anyone make some progress on this?

**Comment by micaelmbagira** [comment]
@snebjorn @codeandcats @bcoe  I have a MR opened https://github.com/conventional-changelog/standard-version/pull/503 which enabled that

**Comment by devjayantmalik** [comment]
> Yes, I'm curious how I can use standard-version in a mono-repo (configured using yarn workspaces, not lerna) consisting of multiple npm packages.
> 
> Like is there a way to tell standard-version to _**only look at git commits that affected files under a certain path**_ when building the change log and bumping the package version?

Is it possible now in 2021 to maintain monorepo with independent versions without lerna? Or do you have some plans for it?

If above comment is possible, then what would be the steps to implement it, a example would be appreciated.

**Comment by phun-ky** [comment]
@micaelmbagira nice work on the PR, but how can we put together this with `standard-version` so we can drop `lerna` completely?

**Comment by ipleten** [comment]
How can I add docs, perf, test and other features to conventional changelog? They are currently just simply being ignored.
